### PR TITLE
Pypi publish

### LINKED ISSUE #96: PyPi release workflow
Create an automated release workflow for SyferText using github actions.

### PR DESCRIPTION
**Pull request by Nilanshrajput**
# Pull Request Template

## Description

Files and workflow for first release of SyferText on PyPi. 

Fixes #96 

## Type of change

Please mark options that are relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

* [x] I did follow the [contribution guidelines](https://github.com/OpenMined/PySyft/blob/master/CONTRIBUTING.md)
* [x] I have commented my code following [Google style](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html).

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        prerelease: false
+
+  deploy:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python
+      uses: actions/setup-python@v1
+      with:
+        python-version: '3.x'
```
Comment (AlanAboudib) [inline .github/workflows/publish.yml:36]: Can you make it '3.6' ?

Comment (Nilanshrajput) [reply .github/workflows/publish.yml]: done

File: setup.py [hunk 2]
```diff
@@ -1,5 +1,10 @@
 from setuptools import setup
 from setuptools import find_packages
+import os
+
+# Utility function function to read README.md file for long description
```
Comment (AlanAboudib) [inline setup.py:5]:
```suggestion
# Utility function to read README.md file for long description
```

File: setup.cfg [hunk 3]
```diff
@@ -0,0 +1,2 @@
+[metadata]
+description-file = README.md
```
Comment (AlanAboudib) [inline setup.cfg:2]: is this variable used somewhere else?

File: .github/workflows/publish.yml [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+      id: create_release
+      uses: actions/create-release@v1
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }} # This token is provided by Actions, you do not need to create your own token
+      with:
+        tag_name: ${{ github.ref }}
+        release_name: Release ${{ github.ref }}
+        body: |
+          Changes in this Release
+        draft: false
```
Comment (AlanAboudib) [inline .github/workflows/publish.yml:26]: Is it normal that the body contains only this string 'Changes in this Release' ?

File: setup.cfg (hunk 3 above)
Comment (Nilanshrajput) [reply setup.cfg:2]: I am not 100% sure but I think it's pypi.

**Comment by Nilanshrajput** [comment]
@AlanAboudib  I think I would remove the automation for release, just keep it for the deploy on pypi. For release could be done manually along with creating release notes?

**Comment by Nilanshrajput** [comment]
And I am aware of SyferText on Pypi:(. But we would be using OpenMined name scoping as suggested here https://github.com/OpenMined/.github/issues/11.

**Comment by AlanAboudib** [comment]
Ok, great. Did you check out my comments @Nilanshrajput ?

**Comment by Nilanshrajput** [comment]
@AlanAboudib can you check the changes

File: .github/workflows/publish.yml [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+      with:
+        python-version: '3.6'
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install setuptools wheel twine
+    - name: Build and publish
+      env:
+        TWINE_USERNAME: ${{ secrets.PYPI_USERNAME }}
+        TWINE_PASSWORD: ${{ secrets.PYPI_PASSWORD }}
```
Comment (AlanAboudib) [inline .github/workflows/publish.yml:23]: Where did we obtain those credentials?

Comment (Nilanshrajput) [reply .github/workflows/publish.yml:23]: You will have to setup your account on pypi, and github let's you store those as secrets.

Comment (Nilanshrajput) [reply .github/workflows/publish.yml:23]: https://help.github.com/en/actions/configuring-and-managing-workflows/creating-and-storing-encrypted-secrets

**Pull request merged**